xbmc/interfaces/legacy/Player.cpp

```diff
@@ -42,12 +42,9 @@ namespace XBMCAddon
     {
       iPlayList = PLAYLIST_MUSIC;
 
-      if (_playerCore == EPC_VideoPlayer ||
-          _playerCore == EPC_MPLAYER ||
-          _playerCore == EPC_PAPLAYER)
-        playerCore = (EPLAYERCORES)_playerCore;
-      else
-        playerCore = EPC_NONE;
+      if (_playerCore != 0)
+        CLog::Log(LOGERROR, "xbmc.Player: Requested non-default player. This behavior is deprecated, plugins may no longer specify a player");
```

Comment:
I'm not sure what you are talking about, I'm just talking about the code in question, here.
It seemed to me that (since you added a log line warning about the deprecated behavior) that you are in fact not aware that the code, as it stands, is backwards incompatible, and will break existing plugins, and that the log line will never be printed.